scripts/common/preflights.sh

```diff
@@ -657,3 +658,18 @@ function bail_if_unsupported_openebs_to_rook_version() {
     fi
 }
 
+# bail_when_requires_minio_with_openebs will bail if OpenEBS and Minio not when it is required
+function bail_when_requires_minio_with_openebs() {
+    if [ -z "$MINIO_VERSION" ] && [ -n "$OPENEBS_VERSION" ]; then
+        if [ -n "$KOTSADM_VERSION" ] && [[ -z "$KOTSADM_DISABLE_S3" || "$KOTSADM_DISABLE_S3" != "1" ]];then
+             bail "OpenEBS and kOTS with s3 enabled requires MinIO"
+        fi
+        if [ -n "$REGISTRY_VERSION" ]  && [ "$HA_CLUSTER" = "1" ];then
+             bail "OpenEBS and Registry for HA installs requires MinIO. Please, ensure that your installer also provides MinIO"
+        fi
+        if [ -n "$VELERO_VERSION" ];then
+             bail "OpenEBS and Valero requires MinIO. Please, ensure that your installer also provides MinIO"
+        fi
```

Comment:
we allow this if s3 is disabled in kots.

```suggestion
        if [ -n "$VELERO_VERSION" ] && [ "$KOTSADM_DISABLE_S3" != 1 ]; then
             logFail "OpenEBS and Valero requires an object store.
             bail "Please, ensure that your installer also provides an object store with either the MinIO or Rook add-on."
        fi
```